BALSAMIC/models/analysis.py

```diff
@@ -400,89 +419,263 @@ class PanelModel(BaseModel):
+                if remove_suffix:
+                    fastq_names.extend(
+                        [
+                            os.path.basename(fastqs.fwd).replace(".fastq.gz", ""),
+                            os.path.basename(fastqs.rev).replace(".fastq.gz", ""),
+                        ]
+                    )
+                else:
+                    fastq_names.extend(
+                        [
+                            os.path.basename(fastqs.fwd),
+                            os.path.basename(fastqs.rev),
+                        ]
+                    )
+        return fastq_names
+
+    def get_fastq_by_fastq_pattern(self, fastq_pattern: str, fastq_type: str) -> str:
+        """Return fastq file path for requested fastq pair pattern and fastq type: [fwd/rev]."""
+        if fastq_type not in [FastqName.FWD, FastqName.REV]:
+            raise ValueError(
+                f"fastq_type must be either {FastqName.FWD} or {FastqName.REV}, not: {fastq_type}"
+            )
+
+        for sample in self.samples:
+            if fastq_pattern in sample.fastq_info:
+                return (
+                    sample.fastq_info[fastq_pattern].fwd
+                    if fastq_type == FastqName.FWD
+                    else sample.fastq_info[fastq_pattern].rev
+                )
+
+    def get_sample_name_by_type(self, sample_type: str) -> str:
+        """Return sample name for requested sample type."""
+        for sample in self.samples:
+            if sample.type == sample_type:
+                return sample.name
+
+    def get_sample_type_by_name(self, sample_name: str, uppercase: bool = False) -> str:
+        """Return sample type for requested sample name, optionally return it capitalized"""
+        for sample in self.samples:
+            if sample.name == sample_name:
+                if uppercase:
+                    return sample.type.upper()
+                return sample.type
+
+    def get_bam_name_per_lane(self, bam_dir: str, sample_name: str) -> List[str]:
+        """Return list of bam-file names for all fastq_patterns of a sample."""
+        bam_names = []
+        for sample in self.samples:
+            if sample.name == sample_name:
+                bam_names.extend(
+                    [
+                        f"{bam_dir}{sample_name}_align_sort_{fastq_pattern}.bam"
+                        for fastq_pattern in sample.fastq_info
+                    ]
+                )
+        return bam_names
+
+    def get_final_bam_name(
+        self, bam_dir: str, sample_name: str = None, sample_type: str = None
+    ) -> str:
+        """Return final bam name to be used in downstream analysis."""
+
+        if sample_name is None and sample_type is None:
+            raise ValueError(
+                "Either sample_name or sample_type must be provided to get the final bam name."
+            )
+
+        if sample_name is None:
+            sample_name = self.get_sample_name_by_type(sample_type)
```

Comment:
Maybe this has to do with how our brains work but i find it easier to follow like this, feel free to disagree
```suggestion
        sample_name = (
            self.get_sample_name_by_type(sample_type) if not sample_name else None
        )
```

Reply:
It honestly doesn't matter much to me, I can change it. It makes sense for me to adapt to have a common style in the code base. 👍

Reply:
Or I guess not. I thought I understood what this code was doing, but I realise I don't. I mean at this point sample_name should never be None. But I think it's a typo and you meant to write if not sample_name else sample_name. I'll implement it with this change : )

Reply:
👍